numpy/_core/src/multiarray/ctors.c

```diff
@@ -2476,7 +2476,9 @@ check_or_clear_and_warn_error_if_due_to_copy_kwarg(PyObject *kwnames)
         Py_XDECREF(traceback);
         if (DEPRECATE("__array__ implementation doesn't accept a copy keyword, "
                       "so passing copy=False failed. __array__ must implement "
-                      "'dtype' and 'copy' keyword arguments.") < 0) {
+                      "'dtype' and 'copy' keyword arguments. "
+                      "To learn more, see the migration guide "
+                      "https://numpy.org/devdocs/numpy_2_0_migration_guide.html#adapting-to-changes-in-the-copy-keyword") < 0) {
```

Comment:
A 128 character line is hideous :) And there is no guarantee that it will print without breaking.

Reply:
we can remove the "#" if you want?

I didn't really know there was a migration guide until i asked https://github.com/numpy/numpy/issues/25941#issuecomment-2336522946 and took some of your time...

Reply:
Just break the line with quotes. Python combines all quoted bits into one string.

Reply:
oh you mean hideous in the code, not hideous for the end user.

Reply:
i made the mod, but i don't think the OSX error is related right?